apps/hpc-ftsadmin/src/app/utils/parse-filters.ts

```diff
@@ -150,43 +173,49 @@ export const decodeFilters = <T extends Filters>(
   initialValues: T
 ): T => {
   try {
-    const res: T = parseInInitialValues<T>(
+    const res: T = parseInInitialValues(
       JSON.parse(stringFilters),
       initialValues
     );
     return res;
   } catch (error) {
     console.warn(
       error,
-      'Error parsing query to JSON. Reseting to initial Values...'
+      'Error parsing query to JSON. Resetting to initial Values...'
     );
     return initialValues;
   }
 };
 
-export function isKey<T extends object>(x: T, k: PropertyKey): k is keyof T {
-  return k in x;
+export function isKey<T>(x: T, k: PropertyKey): k is keyof T {
+  return typeof x === 'object' && x !== null && k in x;
 }
 
-function isFlowObjectTypes(value: string): value is FlowObjectTypes {
-  return [
-    'location',
-    'emergency',
-    'globalCluster',
-    'organization',
-    'plan',
-    'project',
-    'usageYear',
-  ].includes(value);
+const FLOW_OBJECT_TYPES = [
+  'location',
+  'emergency',
+  'globalCluster',
+  'governingEntity',
+  'organization',
+  'anonymizedOrganization',
+  'plan',
+  'project',
+  'usageYear',
+] as const;
+
+export type FlowObjectTypes = (typeof FLOW_OBJECT_TYPES)[number];
+
+export function isFlowObjectTypes(value: string): value is FlowObjectTypes {
+  return new Set<string>(FLOW_OBJECT_TYPES).has(value);
 }
```

Comment:
Since this is being called in a loop, we're creating a new `Set` each time, which is suboptimal. Instead, create the `Set` outside. I guess that you created `Set` to resolve the typing issue that arises when using array `includes()` and thus you put `<string>` generic around `Set` constructor, to avoid typing problems when using `has()`.

If you don't create a `Set` outside, you can cast like this. Unfortunately, there is no way around casting array or `value`.
```suggestion
export const isFlowObjectTypes = (value: string): value is FlowObjectTypes =>
  (FLOW_OBJECT_TYPES as readonly string[]).includes(value);
```